### PR TITLE
core/exec-credential: several cleanups

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/core/exec-credential.c [hunk 1]
```diff
@@ -323,42 +323,42 @@ static int write_credential(
...
+
+                        r = RET_NERRNO(fchown(fd, uid, gid));
+                        if (r < 0)
+                                goto fail;
                 }
         }
 
         if (renameat(dfd, tmp, dfd, id) < 0)
-                return -errno;
+                goto fail;
```
Comment (yuwata) [inline src/core/exec-credential.c]: `r` is not assigned.

**Review by yuwata — APPROVED** [review]
LGTM, except for the one below.

File: src/core/exec-credential.c [hunk 2]
```diff
@@ -775,11 +773,11 @@ static int acquire_credentials(
                  * propagate a credential passed to us from further up. */
 
                 if (path_is_absolute(lc->path)) {
-                        sub_fd = open(lc->path, O_DIRECTORY|O_CLOEXEC|O_RDONLY);
```
Comment (bluca) [inline src/core/exec-credential.c:778]: why is read only dropped here?

Comment (YHNdnzj) [reply src/core/exec-credential.c:778]: If O_DIRECTORY is specified, the mode is effectively unused? I've removed O_RDONLY at several other places like this.

**Pull request merged**